src/Morphir/Scala/Feature/Codec.elm

```diff
@@ -521,7 +521,7 @@ mapTypeDefinitionToDecoder currentPackagePath currentModulePath ( typeName, acce
                         |> ResultList.keepFirstError
 
                 downApply =
-                    hCursor ++ ".downN(0)" ++ ".as(morphir.sdk.string.Codec.decodeString)" ++ ".flatMap"
+                    hCursor ++ """.withFocus(_.withString(str => io.circe.Json.arr (io.circe.Json.fromString(str))))""" ++ ".downN(0)" ++ ".as(morphir.sdk.string.Codec.decodeString)" ++ ".flatMap"
```

Comment:
I think this deserves some explanation. Also, why are we concatenating strings? This should be composed out of Scala AST nodes.

Reply:
About concatenating strings I would have to refactor that. HCursor has method withFocus which gives you the ability to  modify  the focus using a function .  The focus is the particular custom type which we are currently trying to decode.  The method withString picks out the constructors with no arguments which are represented as strings  and converts them to an array.